utility/utils.py

```diff
@@ -1273,3 +1273,20 @@ def method_should_succeed(function, *args, **kwargs):
     log.debug(f"The {function} return status is {rc}")
     if not rc:
         raise AssertionError(f"Execution failed at function {function}")
+
+
+def clone_the_repo(config, node, path_to_clone):
+    """
+    clone the repo on to test node
+    :param config: test config
+    :param node: ceph rgw node
+    :param path_to_clone: the path to clone the repo
+
+    """
+    log.info("cloning the repo")
+    branch = config.get("branch", "master")
+    log.info(f"branch: {branch}")
+    repo_url = config.get("git-url")
+    log.info(f"repo_url: {repo_url}")
+    git_clone_cmd = f"sudo git clone {repo_url} -b {branch}"
+    node.exec_command(cmd=f"cd {path_to_clone} ; {git_clone_cmd}")
```

Comment:
please consider above comment of dynamic directory creation.

Reply:
Added a TODO